src/silo/database.cpp

```diff
@@ -496,16 +493,79 @@ void silo::Database::preprocessing(
    }
 
    SPDLOG_INFO("preprocessing - building database");
+
    build(
       preprocessing_config.sorted_partition_folder.relative_path(),
       preprocessing_config.metadata_file.extension(),
       preprocessing_config.sequence_file.extension(),
-      partition_descriptor,
-      database_config_
+      partition_descriptor
    );
 }
-silo::Database::Database() = default;
 
-std::string silo::buildChunkName(unsigned int partition, unsigned int chunk) {
+void Database::initializeColumns() {
+   for (const auto& item : database_config.schema.metadata) {
+      const auto column_type = item.getColumnType();
+      if (column_type == config::ColumnType::INDEXED_STRING) {
+         storage::column::IndexedStringColumn column;
+         for (size_t i = 0; i < partitions.size(); ++i) {
+            partitions[i].insertColumn(item.name, column.createPartition());
+         }
+         indexed_string_columns.emplace(item.name, std::move(column));
+      } else if (column_type == config::ColumnType::STRING) {
+         storage::column::StringColumn column;
+         for (size_t i = 0; i < partitions.size(); ++i) {
+            partitions[i].insertColumn(item.name, column.createPartition());
+         }
+         string_columns.emplace(item.name, std::move(column));
+      } else if (column_type == config::ColumnType::INDEXED_PANGOLINEAGE) {
+         storage::column::PangoLineageColumn column;
+         for (size_t i = 0; i < partitions.size(); ++i) {
+            partitions[i].insertColumn(item.name, column.createPartition());
+         }
+         pango_lineage_columns.emplace(item.name, std::move(column));
+      } else if (column_type == config::ColumnType::DATE) {
+         auto column = item.name == database_config.schema.date_to_sort_by
+                          ? storage::column::DateColumn(true)
+                          : storage::column::DateColumn(false);
+         for (size_t i = 0; i < partitions.size(); ++i) {
+            partitions[i].insertColumn(item.name, column.createPartition());
+         }
+         date_columns.emplace(item.name, column);
+      } else if (column_type == config::ColumnType::INT) {
+         storage::column::IntColumn column;
+         for (size_t i = 0; i < partitions.size(); ++i) {
+            partitions[i].insertColumn(item.name, column.createPartition());
+         }
```

Comment:
Clang Tidy suggests to use range-based loops here: `for (auto & partition : partitions)`